### DISCUSSION TITLE
Query models that are related to one of the given models

### DISCUSSION THREAD

**Discussion by NiroDeveloper** [laravel/framework#51830] [Ideas]
We all know that it is simple possible to receive models related to a specific model, by the relationship function.
`$user->posts()->get()`

For me, however, the option to search for models that are linked to one of the given models is missing in laravel.
This can be implemented like this.
```
$users = collect([User::create(), User::create(), User::create()]);
$primaryKeyName = $users->first()->getKeyName()
$userIDs = $users->pluck($primaryKeyName);
Posts::whereHas("users", fn(Builder $query) => $query->whereIn($primaryKeyName, $userIDs))->get();
```

I have a feature wish for a new relationship query function, to query models that are related to one of the given models.
```
$users = collect([User::create(), User::create(), User::create()])
Posts::whereRelatedTo("users", $users)->get()
```

**Comment by wizzymore** [comment]
Well you can write it like:
```php
$users = collect([User::create(), User::create(), User::create()])
Posts::whereHas("users", fn($q) => $q->whereIn('id', $users->pluck('id')))->get()
```

Or even something like, and make it reusable:
```php

function relatedTo(Collection $related) {
    if ($related->isEmpty()) {
        return null;
    }
    $key = $related->first()->getKeyName();
    return fn($q) => $q->whereIn($key, $related->pluck($key));
}


$users = collect([User::create(), User::create(), User::create()])
Posts::whereHas("users", relatedTo($users))->get()
```

**Reply by wizzymore** [reply]
Also i don't understand why you don't fetch the users with the posts directly and the pluck and flatten the posts, like:
```php
App\Models\User::with('posts')->get()->pluck('posts')->flatten()
```

This will do the same as what I showed above but I find this to be a more pleasant approach.
